interfaces/builtin/unity8.go

```diff
@@ -0,0 +1,190 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+var unity8ConnectedPlugAppArmor = []byte(`
+# Description: Can access unity8 desktop services
+
+  #include <abstractions/dbus-session-strict>
+
+  # Fonts
+  #include <abstractions/fonts>
+  /var/cache/fontconfig/   r,
+  /var/cache/fontconfig/** mr,
+
+  # Accessibility
+  #include <abstractions/dbus-accessibility-strict>
+  dbus (send)
+       bus=session
+       path=/org/a11y/bus
+       interface=org.a11y.Bus
+       member=GetAddress
+       peer=(name=org.a11y.Bus,label=unconfined),
+  dbus (receive, send)
+       bus=accessibility
+       path=/org/a11y/atspi/**
+       peer=(label=unconfined),
+
+  # URL dispatcher. All apps can call this since:
+  # a) the dispatched application is launched out of process and not
+  #    controllable except via the specified URL
+  # b) the list of url types is strictly controlled
+  # c) the dispatched application will launch in the foreground over the
+  #    confined app
+  dbus (send)
+       bus=session
+       path=/com/canonical/URLDispatcher
+       interface=com.canonical.URLDispatcher
+       member=DispatchURL
+       peer=(name=com.canonical.URLDispatcher,label=###SLOT_SECURITY_TAGS###),
+
+  # This is needed when the app is already running and needs to be passed in
+  # a URL to open. This is most often used with content-hub providers and
+  # url-dispatcher, but is actually supported by Qt generally.
+  dbus (receive)
+       bus=session
+       path=/###PLUG_DBUS_APPIDS###
+       interface=org.freedesktop.Application
+       member=Open
+       peer=(label=###SLOT_SECURITY_TAGS###),
+
+  # Unity launcher (e.g. app counter, progress, alert)
+  dbus (receive, send)
+       bus=session
+       path=/com/canonical/Unity/Launcher/###PLUG_DBUS_APPIDS###
+       peer=(name=com.canonical.Unity.Launcher,label=###SLOT_SECURITY_TAGS###),
+
+  # Content Hub (pasteboard, file transfers)
+  dbus (receive, send)
+       bus=session
+       interface=com.ubuntu.content.dbus.Service
+       path=/
+       peer=(name=com.ubuntu.content.dbus.Service,label=###SLOT_SECURITY_TAGS###),
+
+  # Lttng tracing is very noisy and should not be allowed by confined apps.
+  # Can safely deny. LP: #1260491
+  deny /{,var/}{dev,run}/shm/lttng-ust-* r,
```

Comment:
Let's make this rule more precise (there is no such thing as /var/dev/shm):
```
deny /{dev,run,var/run}/shm/lttng-ust-* r,
```

I think you just copied that from unity7; I'll fix the unity7 interface.